src/main/kotlin/com/zepben/evolve/database/sqlite/metrics/MetricsDataSourceWriter.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2024 Zeppelin Bend Pty Ltd
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package com.zepben.evolve.database.sqlite.metrics
+
+import com.zepben.evolve.database.sqlite.common.SchemaUtils
+import com.zepben.evolve.database.sqlite.common.TableVersion
+import com.zepben.evolve.database.sqlite.extensions.configureBatch
+import com.zepben.evolve.metrics.IngestionJob
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.sql.Connection
+import javax.sql.DataSource
+
+/**
+ * Class for writing metrics to an arbitrary datasource.
+ */
+class MetricsDataSourceWriter @JvmOverloads constructor(
+    private val dataSource: DataSource,
+    private val databaseTables: MetricsDatabaseTables = MetricsDatabaseTables(),
+) {
```

Comment:
discussion: @gcarwardine and I have been playing with changing how we do the constructor injection for testing. Rather than having an overloaded constructor which is really only designed for test injection, you create an internal constructor for the test injection, and a public constructor that takes only the parts you expect users to use. e.g.

```suggestion
class MetricsDataSourceWriter internal constructor(
    private val dataSource: DataSource,
    private val databaseTables: MetricsDatabaseTables,
) {

    constructor(dataSource: DataSource) : this(dataSource, MetricsDatabaseTables())

```

Thoughts?